src/nvim/syntax.c

```diff
@@ -7003,6 +7003,8 @@ highlight_color (
       n = HL_TABLE()[id - 1].sg_cterm_fg - 1;
     else
       n = HL_TABLE()[id - 1].sg_cterm_bg - 1;
+    if (n < 0)
+      return NULL;
```

Comment:
in a separate "lint" commit, please add `{...}` around this block. The linter "cascades", so it will also check up to line 7002 (just add `{...}` to those blocks, too)